src/Api/Jobs/JobsHostedService.cs

```diff
@@ -87,5 +93,6 @@ public static void AddJobsServices(IServiceCollection services, bool selfHosted)
         services.AddTransient<ValidateUsersJob>();
         services.AddTransient<ValidateOrganizationsJob>();
         services.AddTransient<ValidateOrganizationDomainJob>();
+        services.AddTransient<EmptySecretsManagerTrashJob>();
```

Comment:
This will throw an unhandled exception when running the server in `OSS` mode since `EmptySecretsManagerTrashJob` depends on DI from the `SecretRepository` that isn't available for the OSS build.

To replicate run API via `dotnet run -p:Configuration=Debug -p:DefineConstants="OSS"`

We will probably want to extend `JobsHostedService` with a new method `AddCommercialJobsServices` and call it in the else block here https://github.com/bitwarden/server/blob/a565b19ffbe5a1bac7ba0d7b81536dbf6478ddea/src/Api/Startup.cs#L144-L150

Reply:
Thank you for pointing this out! Fixed in [b477147](https://github.com/bitwarden/server/pull/2863/commits/b477147c8286d1a2266ee381ece4894f65cbdb0d)